cli/collector.go

```diff
@@ -0,0 +1,131 @@
+	&cli.IntFlag{Name: "frequency", Usage: "Frequency of the collector execution in days. Values must be between 1 and 30. To be executed monthly it must be filled with '30'"},
+	&cli.IntFlag{Name: "startDay", Usage: "Day of the month for the collector execution. Values must be between 1 and 30"},
+	&cli.IntFlag{Name: "limitMonthBackward", Usage: "The limit month to which the collector must be executed in its historical execution"},
+	&cli.IntFlag{Name: "limitYearBackward", Usage: "The limit year until which the collector must be executed in its historical execution"},
+}
+
+var flagsAddCollectorFromFile = []cli.Flag{
+	altsrc.NewStringFlag(&cli.StringFlag{Name: "id", Usage: "Initials entity like 'trt13'"}),
+	altsrc.NewStringFlag(&cli.StringFlag{Name: "entity", Usage: "Entity from which the collector extracts data like 'Tribunal Regional do Trabalho 13° Região'"}),
+	altsrc.NewStringFlag(&cli.StringFlag{Name: "city", Usage: "City of the entity from which the collector extracts data"}),
+	altsrc.NewStringFlag(&cli.StringFlag{Name: "fu", Usage: "Federation unit of the entity from which the collector extracts data"}),
+	altsrc.NewStringFlag(&cli.StringFlag{Name: "path", Usage: "Collector repository path. Using the import pattern in golang like 'github.com/dadosjusbr/coletores/trt13'"}),
+	altsrc.NewIntFlag(&cli.IntFlag{Name: "frequency", Usage: "Frequency of the collector execution in days. Values must be between 1 and 30. To be executed monthly it must be filled with '30'"}),
+	altsrc.NewIntFlag(&cli.IntFlag{Name: "startDay", Usage: "Day of the month for the collector execution. Values must be between 1 and 30"}),
+	altsrc.NewIntFlag(&cli.IntFlag{Name: "limitMonthBackward", Usage: "The limit month to which the collector must be executed in its historical execution"}),
+	altsrc.NewIntFlag(&cli.IntFlag{Name: "limitYearBackward", Usage: "The limit year until which the collector must be executed in its historical execution"}),
+	&cli.StringFlag{Name: "file", Usage: "File with collector data", Required: true},
+}
+
+// GetCommandAddCollector return the struct for add-collector command
+func GetCommandAddCollector() []*cli.Command {
+	command := []*cli.Command{
+		{
+			Name:   "add-collector",
+			Usage:  "Register a collector from parameters",
+			Action: addCollector,
+			Flags:  flagsAddCollector,
+			Subcommands: []*cli.Command{
+				{
+					Name:   "from-file",
+					Usage:  "Register a collector from a JSON file",
+					Action: addCollector,
+					Before: altsrc.InitInputSourceWithContext(flagsAddCollectorFromFile, altsrc.NewJSONSourceFromFlagFunc("file")),
+					Flags:  flagsAddCollectorFromFile,
+				},
+			},
+		},
+	}
+
+	return command
+
+}
+
+func addCollector(c *cli.Context) error {
+	newCollector, err := parseCollectorFromContext(c)
+	if err != nil {
+		return err
+	}
+	if err := storage.InsertCollector(newCollector); err != nil {
```

Comment:
O insertCollector pode ser "mockado". Para um tutorial sobre mock methods em Go, ver [aqui](https://blog.myhro.info/2018/06/how-to-mock-golang-methods). 

É um tópico confuso, o que acha da gente fazer programação e tentar implementar testes?

Reply:
Conversado na sync semanal.